evm/src/contracts/CircleIntegration/Governance.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.22;
+
+import {ERC1967Upgrade} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+import {BytesParsing} from "src/libraries/BytesParsing.sol";
+import {IWormhole} from "src/interfaces/IWormhole.sol";
+import {IGovernance} from "src/interfaces/IGovernance.sol";
+
+import {State} from "./State.sol";
+import {
+    getRegisteredEmitters,
+    getChainToDomain,
+    getConsumedVaas,
+    getDomainToChain
+} from "./Storage.sol";
+
+abstract contract Governance is IGovernance, State, ERC1967Upgrade {
+    using BytesParsing for bytes;
+
+    /**
+     * @dev Governance emitter chain ID.
+     */
+    uint16 constant _GOVERNANCE_CHAIN = 1;
+
+    /**
+     * @dev Governance emitter address.
+     */
+    bytes32 constant _GOVERNANCE_EMITTER =
+        0x0000000000000000000000000000000000000000000000000000000000000004;
+
+    /**
+     * @dev "CircleIntegration" (left-padded with zeros).
+     */
+    bytes32 constant GOVERNANCE_MODULE =
+        0x000000000000000000000000000000436972636c65496e746567726174696f6e;
+
+    /**
+     * @dev Governance action ID indicating decree to register new emitter and CCTP domain.
+     */
+    uint8 constant _ACTION_REGISTER_EMITTER_AND_DOMAIN = 2;
+
+    /**
+     * @dev Governance action ID indicating decree to upgrade contract.
+     */
+    uint8 constant _ACTION_CONTRACT_UPGRADE = 3;
+
+    /// @inheritdoc IGovernance
+    function registerEmitterAndDomain(bytes memory encodedVaa) public {
```

Comment:
Any reason why you removed the input bytes length checks from before? I don't think there's anything malicious you could do with a weird byte array but a liked the defence in depth